### PR TITLE
Update message formats to match documentation

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [ ] Bug Fix
- [x] Chore
- [ ] Feature
- [ ] Documentation Update
- [ ] Code style update (formatting, local variables)
- [ ] Breaking Change (fix or feature that would cause existing functionality to change)

## Linked tickets

https://digicatapult.atlassian.net/browse/VR-235

## High level description

Modifies query format to be more similar to that in the veritable documentation

## Detailed description

Note this does not completely fix the format of the DRPC messages. In particular I have not yet:

* implemented `createdTime` or `expiresTime`
* removed `quantity`

These will happen in a future PR.

Also fixes several bugs:

* Fix public url on Bob and Charlie personas in docker compose
* Fix mix up between scope3 request/response views
* Fix status bug on query list and credential list pages

## Describe alternatives you've considered

I could have left this more loosely typed and not fixed the above issues. I passed on that

## Operational impact

This will invalidate all existing queries and delete them from deployed environments. As we only have the demo environment this should be fine

## Additional context

None

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -136,7 +136,7 @@ services:
       - DB_NAME=veritable-ui
       - API_SWAGGER_TITLE=Bob
       - API_SWAGGER_BG_COLOR=#ff3131
-      - PUBLIC_URL=${VERITABLE_BOB_PUBLIC_URL:-http://localhost:3000}
```
Comment (Ellenn-A) [inline docker-compose.yml:139]: some interesting stuff we have on main rn

Comment (mattdean-digicatapult) [reply docker-compose.yml:139]: ...interesting

File: src/controllers/queries/__tests__/helpers.ts [hunk 2]
```diff
@@ -60,37 +60,39 @@ const defaultOptions: QueryMockOptions = {
...
       {
         id: 'ccaaaaaa-0000-0000-0000-d8ae0805059e',
         connection_id: mockIds.connectionId,
-        query_type: 'Scope 3 Carbon Consumption',
+        type: 'total_carbon_embodiment',
         status: 'pending_their_input',
         details: {
-          productIds: 'test-1',
-          quantities: '10',
-          connectionIds: 'cccccccc-0000-0000-0000-d8ae0805059e',
```
Comment (Ellenn-A) [inline src/controllers/queries/__tests__/helpers.ts:90]: was this a duplicate of the `connection_id `on line 87?

File: src/controllers/queries/__tests__/index.test.ts [hunk 3]
```diff
@@ -170,7 +171,8 @@ describe('QueriesController', () => {
               action: 'success',
               partialQuery: ['on'],
               productIds: ['product-1', 'product-2'],
-              quantities: ['10', '20'],
+              quantities: [10, 20],
+              emissions: 10,
```
Comment (Ellenn-A) [inline src/controllers/queries/__tests__/index.test.ts:175]: if this wasn't in before, what does it do?

Comment (mattdean-digicatapult) [reply src/controllers/queries/__tests__/index.test.ts:175]: This is our contribution to the emissions which should be added to whatever comes from the cascaded queries

File: src/controllers/queries/index.ts [hunk 4]
```diff
@@ -278,12 +261,12 @@ export class QueriesController extends HTMLController {
     body: {
       companyId: UUID
       action: 'success'
-      emissions?: string
-      partialQuery?: 'on'[]
-      partialSelect?: 'on'[]
+      emissions: number
+      partialQuery?: 'on'[] // TODO: remove
```
Comment (Ellenn-A) [inline src/controllers/queries/index.ts:265]: when should this be removed?

File: src/controllers/queries/index.ts [hunk 5]
```diff
@@ -301,61 +284,58 @@ export class QueriesController extends HTMLController {
...
-        })
-      )
-
-      return this.html(
-        this.scope3CarbonConsumptionResponseTemplates.newScope3CarbonConsumptionResponseFormPage({
-          formStage: 'success',
-          company: connection,
-          query: queryRow,
+          req.log.debug('submitting DRPC request to %s connection', connectionIds[i])
+          // TODO: handle errors
```
Comment (Ellenn-A) [inline src/controllers/queries/index.ts:311]: will be done in another ticket?

File: src/controllers/queries/index.ts [hunk 6]
```diff
@@ -401,120 +373,147 @@ export class QueriesController extends HTMLController {
...
-          query_type: 'Scope 3 Carbon Consumption',
-          status: 'pending_their_input',
-          details: localQuery,
-          response_id: null,
-          query_response: null,
-          role: 'requester',
-        })
-        if (!payload.query) payload.query = query
+  private async handleQueryRequest(
+    log: Logger,
```
Comment (Ellenn-A) [inline src/controllers/queries/index.ts:377]: @dblane-digicatapult say we need more logging

File: src/services/drpcEvents/index.ts [hunk 7]
```diff
@@ -197,26 +183,26 @@ export default class DrpcEvents {
       }
       //find corresponding query based on queryIdForResponse provided by responder,
       //this should match a query id in our db
-      const [queryRow] = await this.db.get('query', { id: params.queryIdForResponse, role: 'requester' })
+      const [queryRow] = await this.db.get('query', { id: params.id, role: 'requester' })
```
Comment (Ellenn-A) [inline src/services/drpcEvents/index.ts:186]: this rename to it just being an 'id' is a bit confusing ngl...cuz this is id not the local request it, it's still the request id from the requester, right?

File: src/controllers/queries/index.ts (hunk 6 above)
Comment (mattdean-digicatapult) [reply src/controllers/queries/index.ts:377]: agreed next PR I'll fill in the gaps

**Comment by mattdean-digicatapult** [comment]
@Ellenn-A 


Uploading Screen Recording 2024-10-30 at 19.43.46.mov…

File: src/services/drpcEvents/index.ts (hunk 7 above)
Comment (mattdean-digicatapult) [reply src/services/drpcEvents/index.ts:186]: this is just to match the format set by @doowt in the docs. I think it's ok as long as seen in the proper context

**Review by Ellenn-A — APPROVED** [review]
yep tested this morning

**Pull request merged**